drivers/storage/portworx/components.go

```diff
@@ -2012,6 +2059,184 @@ func (p *portworx) createPortworxAPIDaemonSet(
 	return k8sutil.CreateOrUpdateDaemonSet(p.k8sClient, newDaemonSet, ownerRef)
 }
 
+func (p *portworx) createServiceMonitor(
+	cluster *corev1alpha1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	svcMonitor := &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pxServiceMonitor,
+			Namespace: cluster.Namespace,
+			Labels: map[string]string{
+				"name": pxServiceMonitor,
+			},
+			OwnerReferences: []metav1.OwnerReference{*ownerRef},
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Endpoints: []monitoringv1.Endpoint{
+				{
+					Port: pxRESTPortName,
+				},
+				{
+					Port: pxKVDBPortName,
```

Comment:
We should only add the Kvdb endpoint if we are using internal ETCD.
I don't know how if this is always the case. But if they can use an external etcd these will show up as "n"  failing endpoints on the ui.

Reply:
Done

Reply:
Looks good